typing/typecore.ml

```diff
@@ -1410,14 +1414,19 @@ and type_pat_aux ~constrs ~labels ~no_existentials ~mode ~explode ~env
       k { p with pat_extra =
         (Tpat_type (path, lid), loc, sp.ppat_attributes) :: p.pat_extra }
   | Ppat_open (lid,p) ->
-      let path, new_env =
-        !type_open Asttypes.Fresh !env sp.ppat_loc lid in
-      let new_env = ref new_env in
-      type_pat ~env:new_env p expected_ty ( fun p ->
-        env := Env.copy_local !env ~from:!new_env;
-        k { p with pat_extra =( Tpat_open (path,lid,!new_env),
-                            loc, sp.ppat_attributes) :: p.pat_extra }
-      )
+      let me = Ast_helper.Mod.ident ~loc:lid.loc ~attrs:[] lid in
+      let id, _tme, new_env =
+        !type_open Asttypes.Fresh !env sp.ppat_loc me in begin
+      match id with
+      | Some _ -> assert false
```

Comment:
Why is this case impossible?  Is that because of syntactic restrictions on local open in patterns?

Reply:
documented and checked in ast_invariants